dbcon/joblist/groupconcat.cpp

```diff
@@ -606,27 +635,7 @@ int64_t GroupConcator::lengthEstimate(const rowgroup::Row& row)
             case CalpontSystemCatalog::DECIMAL:
             case CalpontSystemCatalog::UDECIMAL:
             {
-                int64_t v = row.getIntField(*i);
-                double scale = row.getScale(*i);
-
-                if (scale > 0)
-                {
-                    v /= (int64_t) pow(10.0, scale);
-
-                    if (v < 0) fieldLen++;
-
-                    while ((v /= 10) != 0) fieldLen++;
-
-                    fieldLen += (int64_t) scale + 2;;
-                }
-                else
-                {
-                    if (v < 0) fieldLen++;
-
-                    while ((v /= 10) != 0) fieldLen++;
-
-                    fieldLen += 1;;
-                }
+                fieldLen += row.getPrecision(*i)/2;
```

Comment:
Why are we dividing by 2 here?

Reply:
To simplify expected row length calculation. Why I took / 2. This is an assumption that every element of a concatenated string uses half of its precision. However now I see it should be at least scale + half of the prec - scale.
I've also created an issue to refactor group_concat code removing length estimation.

Reply:
Originally, we were calculating the fieldLen as the complete length of the DECIMAL here, now we are setting it to 1. But it looks like for DOUBLE/FLOAT fields as well, we are setting fieldLen=1. I am not sure why.  I obviously don't have the full context on this function.

Reply:
I bet this is because of complexity of the actual length calculationfor floats and doubles. We will optimize out this length estimation in future.